homeassistant/components/jenkins/config_flow.py

```diff
@@ -0,0 +1,103 @@
+"""Config flow to configure the Jenkins integration."""
+from jenkinsapi.jenkins import Jenkins
+from requests.exceptions import HTTPError, MissingSchema
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.jenkins import _LOGGER
+from homeassistant.components.jenkins.const import CONF_JOB_NAME, DOMAIN
+from homeassistant.const import CONF_HOST, CONF_TOKEN, CONF_USERNAME
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class JenkinsFlowHandler(config_entries.ConfigFlow):
+    """Handle Jenkins config flow."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize flow."""
+        self.url = None
+        self.username = None
+        self.token = None
+        self.server = None
+        self.job_name = None
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        data_schema = vol.Schema(
+            {
+                vol.Required(CONF_HOST): str,
+                vol.Optional(CONF_USERNAME): str,
+                vol.Optional(CONF_TOKEN): str,
+            }
+        )
+
+        if user_input is not None:
+            self.url = user_input[CONF_HOST]
+            self.username = (
+                user_input[CONF_USERNAME] if CONF_USERNAME in user_input else None
+            )
+            self.token = user_input[CONF_TOKEN] if CONF_TOKEN in user_input else None
+
+            try:
+                self.server = await self.hass.async_add_executor_job(
+                    Jenkins, self.url, self.username, self.token
+                )
+                _LOGGER.debug("Successfully connected to host: %s", self.url)
+                return await self.async_step_job()
+            except MissingSchema:
```

Comment:
You can simply populate `errors` in all of these cases and eventually have a single call to `self.async_show_form`